File: app/src/main/java/com/ninhttd/devtest/presentation/product/TabFragment.java
package com.ninhttd.devtest.presentation.product;

import android.os.Bundle;
import android.view.View;
import android.widget.LinearLayout;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.ninhttd.devtest.R;
import com.ninhttd.devtest.base.BaseFragment;
import com.ninhttd.devtest.presentation.product.adapter.ThongSoKiThuatAdapter;
import com.ninhttd.devtest.data.entity.AttributeGroup;
import com.ninhttd.devtest.utils.Constant;

import java.util.List;

public class TabFragment extends BaseFragment {
    private static final String TAG = "TabFragment";
    List<AttributeGroup> attributeGroups;


    @Override
    protected void afterView() {
        Bundle arguments = getArguments();
        String tabTitle = arguments.getString(Constant.ViewParam.TAB_TITLE);
        attributeGroups = (List<AttributeGroup>) arguments.getSerializable(Constant.ViewParam.ATTRIBUTE_GROUP);

        if (tabTitle.equals(getContext().getString(R.string.tab_title_0))) {
            initTabMoTa();
        } else if (tabTitle.equals(getContext().getString(R.string.tab_title_1))) {
            initTabThongSoKT();
        } else {
            initTabSoSanhGia();
        }
    }

    private void initTabSoSanhGia() {

    }

    private void initTabThongSoKT() {
        RecyclerView rvThongSoKt = getActivity().findViewById(R.id.rv_thong_so_kt);
        LinearLayout btnShowMore = getActivity().findViewById(R.id.btn_show_more);
        if (attributeGroups.size() > 3) {
            btnShowMore.setVisibility(View.VISIBLE);
        } else {
            btnShowMore.setVisibility(View.GONE);
        }
        ThongSoKiThuatAdapter thongSoKiThuatAdapter = new ThongSoKiThuatAdapter(attributeGroups, getActivity());
        rvThongSoKt.setLayoutManager(new LinearLayoutManager(this.getContext(), RecyclerView.VERTICAL, false));
        rvThongSoKt.setAdapter(thongSoKiThuatAdapter);
        rvThongSoKt.setNestedScrollingEnabled(false);
    }

    private void initTabMoTa() {
    }

    @Override
    protected int getLayout() {
        Bundle arguments = getArguments();
        String tabTitle = arguments.getString(Constant.ViewParam.TAB_TITLE);

        if (tabTitle.equals(getContext().getString(R.string.tab_title_0))) {
            return R.layout.fragment_tab_mo_ta;
        } else if (tabTitle.equals(getContext().getString(R.string.tab_title_1))) {
            return R.layout.fragment_tab_thong_so_ky_thuat;
        } else {
            return R.layout.fragment_tab;
        }
    }

    @Override
    protected void destroyView() {

    }


}

File: app/src/main/java/com/ninhttd/devtest/presentation/product/TabFragment.kt
package com.ninhttd.devtest.presentation.product

import android.view.View
import android.widget.LinearLayout
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.ninhttd.devtest.R
import com.ninhttd.devtest.base.BaseFragment
import com.ninhttd.devtest.data.entity.AttributeGroup
import com.ninhttd.devtest.presentation.product.adapter.ThongSoKiThuatAdapter
import com.ninhttd.devtest.utils.Constant

class TabFragment : BaseFragment() {
    companion object {
        private val TAG = "TabFragment"
    }

    private var attributeGroups: List<AttributeGroup>? = null

    override val layout: Int
        get() {
            val arguments = arguments
            val tabTitle = arguments!!.getString(Constant.ViewParam.TAB_TITLE)

            return when (tabTitle) {
                getString(R.string.tab_title_0) -> R.layout.fragment_tab_mo_ta
                getString(R.string.tab_title_1) -> R.layout.fragment_tab_thong_so_ky_thuat
                else -> R.layout.fragment_tab
            }
        }


    override fun afterView() {
        val arguments = arguments
        val tabTitle = arguments!!.getString(Constant.ViewParam.TAB_TITLE)
        attributeGroups = arguments.getSerializable(Constant.ViewParam.ATTRIBUTE_GROUP) as List<AttributeGroup>

        when (tabTitle) {
            context!!.getString(R.string.tab_title_0) -> initTabMoTa()
            context!!.getString(R.string.tab_title_1) -> initTabThongSoKT()
            else -> initTabSoSanhGia()
        }
    }

    private fun initTabSoSanhGia() {

    }

    private fun initTabThongSoKT() {
        activity?.apply {
            val rvThongSoKt = findViewById<RecyclerView>(R.id.rv_thong_so_kt)
            val btnShowMore = findViewById<LinearLayout>(R.id.btn_show_more)
            if (attributeGroups!!.size > 3) {
                btnShowMore.visibility = View.VISIBLE
            } else {
                btnShowMore.visibility = View.GONE
            }
            val thongSoKiThuatAdapter = ThongSoKiThuatAdapter(attributeGroups, activity!!)
            rvThongSoKt.layoutManager = LinearLayoutManager(this, RecyclerView.VERTICAL, false)
            rvThongSoKt.adapter = thongSoKiThuatAdapter
            rvThongSoKt.isNestedScrollingEnabled = false
        }

    }

    private fun initTabMoTa() {}

    override fun destroyView() {

    }


}
